tensorflow_probability/examples/jupyter_notebooks/Cox_Process_TFP.ipynb

```diff
@@ -0,0 +1,1409 @@
+            "302                    0                    4\n",
+            "303                    0                    0\n",
+            "304                  244                   18\n",
+            "305                   63                  163\n",
+            "306                    5                    6\n",
+            "307                    2                1,543\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "metadata": {
+        "id": "J808jVnDRlWo",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Model\n",
+        "\n",
+        "So for our point process, we want to be able to simulate one fo these $N \\times V$ datasets, which we can do with a latent intensity function $f$. This will be a multivariate normal distribution for which we calculate the covariance for every single NBA player across all of the positions they made shots from (not all players are going to be equal when it comes to which position they're best from).  The outputs of this function will then be fed\n",
+        "\n",
+        "We model a latent intensity function for each data point. Let $K$ be the\n",
+        "$N \\times V \\times V$ covariance matrix applied to the data set $X$ with fixed\n",
+        "kernel hyperparameters, where a slice $K_n$ is the $V \\times V$ covariance\n",
+        "matrix over counts for a data point $x_n$.\n",
+        "\n",
+        "$ \\text{For } n = 1, ..., N$  (with $N$ being the number of players),\n",
+        "\n",
+        "$$ p(f_n) = N(f_n | 0, K_n) \\text{,} $$"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "H5q3GGNU-yiI",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "First we need to define the Radial basis function kernel, also known as the squared exponential or exponentiated quadratic:\n",
+        "\n",
+        "It is defined as $k(x, x') = \\sigma^2 \\exp\\Big(-\\ \\frac{1}{2} \\sum_{d=1}^D \\ \\frac{1}{\\ell_d^2} (x_d - x'_d)^2 \\Big)$ for output variance $\\sigma^2$ and lengthscale $\\ell^2$.\n",
+        "  \n",
+        "The kernel is evaluated over all pairs of rows, `k(X[i, ], X2[j, ])`. If `X2` is not specified, then it evaluates over all pairs of rows in `X`, `k(X[i, ], X[j, ])`. The output is a matrix where each entry (`i`, `j`) is the kernel over the `i`th and `j`th rows.\n",
+        "  "
+      ]
+    },
+    {
+      "metadata": {
+        "id": "CkPfDPvF-URf",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "#@title RBF Function Definition  { display-mode: \"form\" }\n",
+        "from __future__ import absolute_import, division, print_function\n",
+        "\n",
+        "import tensorflow as tf\n",
+        "\n",
+        "\n",
+        "\n",
+        "def rbf(X, X2=None, lengthscale=1.0, variance=1.0):\n",
```

Comment:
could we use the kernels package?
```
tfpk = tfp.positive_semidefinite_kernels
kernel = tfpk.ExponentiatedQuadratic()
cov = kernel.matrix(X, X2)
```
@csuter how would we add jitter? explicitly add eye below?